packages/typespec-vscode/src/typespec-utils.ts

```diff
@@ -198,3 +199,114 @@ export async function installCompilerWithUi(
   }
   return result;
 }
+
+function getConfigEntriesFromEmitterOptions(
+  emitterOptions: Record<string, any>,
+  configYamlContent: string,
+  defaultEmitKeys: string[] = [],
+): Record<string, any> {
+  const configEntries: Record<string, any> = {};
+  if (!emitterOptions.properties) {
+    return configEntries;
+  }
+
+  for (const [propertyName, propertySchema] of Object.entries(emitterOptions.properties)) {
+    if (configYamlContent.includes(propertyName)) {
+      continue;
+    }
+    if (defaultEmitKeys.includes(propertyName)) {
+      continue;
+    }
+
+    const propSchema = propertySchema as any;
+    let comment = "";
+    if (propSchema.description) {
+      comment = propSchema.description;
+    }
+    if (propSchema.type) {
+      comment += ` (Type: ${propSchema.type})`;
+    }
+    if (propSchema.enum) {
+      comment += ` (Options: ${propSchema.enum.join(", ")})`;
+    }
+
+    let defaultValue: any;
+    if (propSchema.default !== undefined) {
+      defaultValue = propSchema.default;
+    }
+
+    configEntries[propertyName] = {
+      value: defaultValue,
+      comment: comment,
+    };
+  }
+
+  return configEntries;
+}
+
+async function extractEmitterOptions(
```

Comment:
i meant to put this method in util, the other 2 is still specific to the emit-code scenario so should be in the emit-code side.